### DISCUSSION TITLE
"Share all files" on, but still need to add a "share: true" property

### DISCUSSION THREAD

**Discussion by WolfieKnee** [ObsidianPublisher/obsidian-github-publisher#270] [Q&A — Plugin] [answered]
Hi - I'm getting inconsistency behaviour when using the "share all files" settings in the plugin: 
- If I add new file to my vault and run a sync it doesn't work
- if I add a share: true property it syncs fine

Am I expecting the correct behaviour or is there potentially something wrong here?

thanks*
Dave

* and thanks for a great tool!

obsidian 1.5.3 on ubuntu 22.X with plugin 6.11.5

**Comment by Mara-Li** [comment]
Please, read the doc!
Unless you toggle the share without key, you need to use the share key!

**Reply by WolfieKnee** [reply]
![Screenshot from 2024-01-08 09-40-23](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/145220120/d314308d-56c4-4f29-86b3-54b288a3a902)

yup - based on the docs I believe that's what I have done and isn't working.

**Reply by Mara-Li** [reply]
Okay, that's strange. Do you have anything in the console when this thing happens ?

**Reply by WolfieKnee** [reply]
I'm running from within the Obsidian GUI, so there are no consoles (I'm happy to do this, or share log files if it helps debug, but I would need to get into the docs for the command/log location).

I am running with the full log GUI - it's a bit quick, but it looks like:
- it finds all the files ok
- it updates existing files
- it doesn't push new files (unless they have the share key set to true)

**Reply by Mara-Li** [reply]
What do you mean by Obsidian GUI ? 
The console is opened by CTRL + MAJ + I on Windows (CMD for mac). I don't know on Linux (I heard the linux version have some problem actually?°

**Reply by WolfieKnee** [reply]
I'm slightly worried we might be talking at cross purposes, but I'd like to work this one out...
- At this time I'm focused just on pushing my vault to gitHub (I'm using it to capture notes on a course, and hope to have time to publish some of them when I'm finished!)
- I'm happy working in the terminal, but have not found any docs on doing that for obsidian.
- I'm using obsidian through the standard interface, and not aware that that has a console.
- when I turn on logs in the plugin I see a lot of good looking info flash past in pop-ups, but it's not there long enough to capture useful input here. _Are those logs available anywhere as a file I can share?_ I've screenshot a few of them here - looks like a lot of `TypeErrors`..
![Screenshot from 2024-01-08 22-08-44](https://github.com/ObsidianPublisher/obsidian-github-publisher/assets/145220120/641f18d6-7c95-40a6-8684-c7363b0ea24e)

thanks!

**Reply by Mara-Li** [reply]
You can open the console of Obsidian using CTRL + MAJ + I.

**Reply by WolfieKnee** [reply]
Thanks - I knew I was missing something! 
Yes, there's a lot in the console:
- console > nothing from this plugin
- Network: lots from this plugin. mostly status 200 from successful fetches, a 201 and one 422 that doesn't seem to be related to any of the skipped files. 

Looking at the paths on the successful commits, the files which are not shared are not there. If I add a share flag on a page then it is added to the commits fine. It seems like the gitHub integration is fine, but the 'share all' is not working locally. 

Is there anything specific it is helpful to share?

PS - the course I'm doing is a dev bootcamp, so if it seems like I've strangely patchy knowledge it is because I do!

**Reply by Mara-Li** [reply]
I think I will make an update for the plugin to get a log file in the folder, but testing on my side, I don't have any problem with sharing without key (all documents was send). Can you export your settings ?

**Comment by Mara-Li** [comment, marked as answer]
See #271 
Fixed in v.6.13.0

**Reply by WolfieKnee** [reply]
Great! I'll get the update installed and confirm back for you...

**Reply by WolfieKnee** [reply]
yup - all good now, thanks so much! Kofi inbound...
